### PR TITLE
chore(PPDSC-2732): migrate staging docs from k8s to S3

### PR DESCRIPTION
**Pull request by JohnTParsons**
PPDSC-2732

**What**

1. Background - this is needed because eks_kubectl_deploy_staging does not work since the CircleCI secrets rotation
2. What did you do - created S3 roles and buckets instead
3. What does the reviewers should expect. Generate docs and storybook for staging, as before, but in new newskit prod a/c (shared by staging and prod envs)

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**How to test:**
Test using
https://ncu-newskit-docs-staging.s3.eu-west-1.amazonaws.com/index.html
https://ncu-newskit-docs-staging.s3.eu-west-1.amazonaws.com/timestamp.html
https://ncu-newskit-docs-staging.s3.eu-west-1.amazonaws.com/storybook/index.html
https://ncu-newskit-docs-staging.s3.eu-west-1.amazonaws.com/storybook/timestamp.html

### CODE REVIEW

**Comment by pp-serviceaccount** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2732-staging-s3
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2732-staging-s3/storybook

**Review by mutebg — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -1798,33 +1841,27 @@ workflows:
             - test_e2e_comps
             - test_e2e_docs_dev
 
+      - push_s3_general_oidc:
+          name: deploy_to_s3_staging
+          context: ncu-newskit-site-stage
+          s3Path_docs: s3://ncu-newskit-docs-staging/
+          s3Path_storybook: s3://ncu-newskit-docs-staging/storybook/
```
Comment (Xin00163) [inline .circleci/config.yml:1848]:
```suggestion
          s3Path_storybook: s3://ncu-newskit-storybook-staging/storybook/
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -2003,30 +2048,31 @@ workflows:
...
             - build_push_docker_ecr_release
-      - eks_kubectl_deploy:
-          <<: *only_on_version_tag
-          name: eks_kubectl_deploy_staging
+      - push_s3_general_oidc:
+          name: deploy_to_s3_staging
           context: ncu-newskit-site-stage
-          environment: staging
+          s3Path_docs: s3://ncu-newskit-docs-staging/
+          s3Path_storybook: s3://ncu-newskit-docs-staging/storybook/
```
Comment (Xin00163) [inline .circleci/config.yml:2062]:
```suggestion
          s3Path_storybook: s3://ncu-newskit-storybook-staging/storybook/
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -1695,6 +1729,15 @@ workflows:
           environment: dev
           requires:
             - init_aws
+      - deploy_terraform_oidc:
+          name: deploy_terraform_s3_staging
+          package: s3/prod-newskit
+          context: ncu-newskit-site-stage
+          apply: true
```
Comment (Xin00163) [inline .circleci/config.yml:1736]: It is `false` in other jobs

File: .circleci/config.yml (hunk 2 above)
Comment (Xin00163) [reply .circleci/config.yml:2062]: I noticed that the bucket is called `"ncu-newskit-docs-staging"`, is storybook supposed to be there?

File: .circleci/config.yml (hunk 3 above)
Comment (JohnTParsons) [reply .circleci/config.yml:1736]: It is false for the "validate_deploy_terraform_s3_*" jobs (to act as a dry run check of the Terraform running init and plan but not apply) BUT it is true for the actual "deploy_terraform_s3_*" jobs, as we want them to apply the Terraform

File: terraform/s3/dev-newskit/dev.tfvars [hunk 4]
```diff
@@ -0,0 +1,3 @@
+environment = "dev"
+tag_name_suffix = "-dev"
+docs_bucket = "ncu-newskit-docs-dev"
```
Comment (Xin00163) [inline terraform/s3/dev-newskit/dev.tfvars:3]: This also includes storybook so do we want to call it "ncu-newskit-dev"?

Comment (JohnTParsons) [reply terraform/s3/dev-newskit/dev.tfvars:3]: The original NGN bucket is called "ncu-newskit", so is based on that naming convention.

(as before /index.html is the docs site and /storybook/index.html is storybook)

Comment (JohnTParsons) [reply terraform/s3/dev-newskit/dev.tfvars:3]: The ncu-newskit-dev bucket was created in last week's PR and is in current use.
This PR adds to ncu-newskit-staging bucket.

**Review by Xin00163 — APPROVED** [review]

**Pull request merged**